addons/website_slides/controllers/main.py

```diff
@@ -458,17 +460,48 @@ def _get_user_slide_authorization(self, slide_id):
         return {'status': status, 'slide': slide, 'channel_id': slide.sudo().channel_id.id}
 
     @http.route([
+        '/slides/<int:channel_id>',
+        '/slides/<int:channel_id>/category/<int:category_id>',
+        '/slides/<int:channel_id>/category/<int:category_id>/page/<int:page>',
         '/slides/<model("slide.channel"):channel>',
         '/slides/<model("slide.channel"):channel>/page/<int:page>',
         '/slides/<model("slide.channel"):channel>/tag/<model("slide.tag"):tag>',
         '/slides/<model("slide.channel"):channel>/tag/<model("slide.tag"):tag>/page/<int:page>',
         '/slides/<model("slide.channel"):channel>/category/<model("slide.slide"):category>',
         '/slides/<model("slide.channel"):channel>/category/<model("slide.slide"):category>/page/<int:page>',
     ], type='http', auth="public", website=True, sitemap=sitemap_slide)
-    def channel(self, channel, category=None, tag=None, page=1, slide_category=None, uncategorized=False, sorting=None, search=None, **kw):
+    def channel(self, channel=False, channel_id=False, category=None, category_id=False, tag=None, page=1, slide_category=None, uncategorized=False, sorting=None, search=None, **kw):
         """
         Will return all necessary data to display the requested slide_channel along with a possible category.
         """
+        partner_id = False
+        is_partner_without_user = False
+        invite_hash = ''
+        is_share_preview = False
+
+        # Invitation data
+        if request.website.is_public_user() and channel_id and not channel:
+            invite_hash = kw.get('invite_hash')
+            partner_id = int(kw.get('partner_id')) or False
+            if partner_id and invite_hash:
+                channel_sudo = request.env['slide.channel'].browse(channel_id).exists().sudo()
+                partner_sudo = request.env['res.partner'].browse(partner_id).exists().sudo()
+                if partner_sudo and channel_sudo.is_published:
+                    partner_hash = tools.hmac(request.env(su=True), 'website_slides-channel-invite', (partner_id, channel_id))
+                    if hmac.compare_digest(partner_hash, invite_hash):
+                        channel_partner = channel_sudo.channel_partner_all_ids.filtered(lambda cp: cp.partner_id.id == partner_id)
+                        if channel_partner.member_status == 'invited':
+                            is_partner_without_user = not partner_sudo.user_ids or all(user._is_public() for user in partner_sudo.user_ids)
+                            is_share_preview = channel_sudo.visibility == 'members'
```

Comment:
I would prefer a more explicit name such as `is_visible_via_invite` (if that *is* what it means)?